src/emoji/EmojiPickerScreen.js

```diff
@@ -0,0 +1,86 @@
+/* @flow */
+import { connect } from 'react-redux';
+
+import React, { Component } from 'react';
+import { View, Text, FlatList } from 'react-native';
+import type { NavigationScreenProp } from 'react-navigation';
+
+import { emojiReactionAdd } from '../api';
+import codePointMap from './codePointMap';
+import { Screen, Input } from '../common';
+import EmojiRow from './EmojiRow';
+import getFilteredEmojiList from './getFilteredEmojiList';
+import type { GlobalState, RealmEmojiState, Auth, Dispatch } from '../types';
+import { getActiveRealmEmoji, getAuth } from '../selectors';
+import { navigateBack } from '../nav/navActions';
+
+type Props = {
+  realmEmoji: RealmEmojiState,
+  auth: Auth,
+  dispatch: Dispatch,
+  navigation: NavigationScreenProp<*> & {
+    state: {
+      params: {
+        messageId: number,
+      },
+    },
+  },
+};
+
+type State = {
+  filter: string,
+};
+
+class EmojiPickerScreen extends Component<Props, State> {
+  props: Props;
+  state: State;
+
+  state = {
+    filter: '',
+  };
+
+  handleInputChange = text => {
+    this.setState({
+      filter: text.toLowerCase(),
+    });
+  };
+
+  addReaction = item => {
+    const { auth, dispatch } = this.props;
+    const { messageId } = this.props.navigation.state.params;
+    emojiReactionAdd(auth, messageId, 'unicode_emoji', codePointMap[item], item);
```

Comment:
Once you've added the types, I am not sure if you'll get an error here, but the parameters look incorrect. If `item` is of type `RealmEmojiType` this code will not be correct.

Reply:
`item` is just a string that is the emoji name.